tests/python/relay/test_pipeline_executor.py

```diff
@@ -0,0 +1,276 @@
+    net_output2 = relay.subtract(data, mv2)
+    net_output3 = relay.multiply(data, mv3)
+
+    """
+    # net2 use net1 output1 as input.
+    """
+    net2 = relay.add(data_net1_output_1, mv2)
+    net2 = relay.add(net2, data21)
+    net2 = relay.add(net2, mv3)
+
+    """
+    # net3 use net2 output1 and net1 outpu2 as input.
+    """
+    net3 = relay.multiply(data_net2_output_1, mv3)
+    net3 = relay.add(net3, data_net1_output_2)
+
+    mods.append(
+        tvm.IRModule.from_expr(
+            relay.Function([data], relay.Tuple([net_output1, net_output2, net_output3]))
+        )
+    )
+    mods.append(tvm.IRModule.from_expr(relay.Function([data_net1_output_1, data21], net2)))
+    mods.append(
+        tvm.IRModule.from_expr(relay.Function([data_net1_output_2, data_net2_output_1], net3))
+    )
+
+    return mods, dshape
+
+
+def get_manual_conf(mods, target):
+    """
+    # This function use to generate manual pipe line configueration,
+    # the result use to verify if the pipe configuration can generate
+    # correct result.
+    """
+    mod_config = {}
+    """
+    # set configure
+    """
+
+    """
+    # third output is final output, second output for mod3, first for mod2
+    # input
+    """
+    pipe_config1 = {
+        "mod_indx": 1,
+        "output": [
+            {"output_indx": 0, "dependent": [{"mod_indx": 2, "input_name": "data_0"}]},
+            {"output_indx": 1, "dependent": [{"mod_indx": 3, "input_name": "data_0"}]},
+            {"output_indx": 2, "dependent": [{"mod_indx": 0, "input_name": "0"}]},
+        ],
+    }
+    mod_config[mods[0]] = {
+        "pipeline": pipe_config1,
+        "target_host": None,
+        "mod_name": "default",
+        "build": None,
+        "params": None,
+        "target": target[0],
+        "dev": target[1],
+    }
+
+    pipe_config2 = {
+        "mod_indx": 2,
+        "output": [
+            {"output_indx": 0, "dependent": [{"mod_indx": 3, "input_name": "data_1"}]},
+        ],
+    }
+    mod_config[mods[1]] = {
+        "pipeline": pipe_config2,
+        "target_host": None,
+        "mod_name": "default",
+        "build": None,
+        "params": None,
+        "target": "llvm",
+        "dev": tvm.cpu(0),
+    }
+
+    pipe_config3 = {
+        "mod_indx": 3,
+        "output": [{"output_indx": 0, "dependent": [{"mod_indx": 0, "input_name": "1"}]}],
+    }
+    mod_config[mods[2]] = {
+        "pipeline": pipe_config3,
+        "target_host": None,
+        "mod_name": "default",
+        "build": None,
+        "params": None,
+        "target": "llvm",
+        "dev": tvm.cpu(0),
+    }
+    return mod_config
+
+
+def pipeline(target):
+    """
+    #Get 3 pipeline module.
+    """
+    (mod1, mod2, mod3), dshape = get_mannual_mod()
+
+    # Prepare batch data for pipeline feeding
+    datas = []
+    for i in range(5):
+        datas.append(np.full(dshape, 3 + i).astype("float32"))
+
+    # Runtime error check
+    pipe_config_check(mod1, mod2, mod3)
```

Comment:
Put this to a separate test. For example, you can have
```python
def test_pipeline_config():
      (mod1, mod2, mod3), dshape = get_mannual_mod()
      # Inline "pipe_config_check" here
```

This check should be target independent so it should not be tested many times. Putting it to a separate test can also help locate the problem when test failure.

Reply:
fixed.